### PR TITLE
Add Htmlable support

### PR DESCRIPTION
**Pull request by megaxlr**
Fixes GrahamCampbell/Laravel-Markdown#91

Signed-off-by: MegaXLR <admin@megaxlr.net>

### CODE REVIEW

File: src/Converters/Converter.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class Converter extends \League\CommonMark\Converter {
+	
+	/**
+	 * Converts CommonMark to HTML.
+	 *
+	 * @param string $commonMark
+	 *
+	 * @return string
+	 *
+	 * @api
```
Comment (GrahamCampbell) [inline src/Converters/Converter.php]: Please don't use the API annotation.

Comment (megaxlr) [reply src/Converters/Converter.php]: That block was autogenerated by Eclipse. Completely looked over that, sorry.
Adressed in 44be4b8316b891964e4c04726fd4bd0d6bd5c2e9

File: src/Converters/Converter.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+use GrahamCampbell\Markdown\Responses\MarkdownString;
+
+class Converter extends \League\CommonMark\Converter {
+	
+	/**
+	 * Converts CommonMark to HTML.
+	 *
+	 * @param string $commonMark
+	 *
+	 * @return string
```
Comment (GrahamCampbell) [inline src/Converters/Converter.php]: The return type is not string, thus violating the interface.

File: src/Converters/Converter.php [hunk 3]
```diff
@@ -10,7 +10,7 @@ class Converter extends \League\CommonMark\Converter {
 	 *
 	 * @param string $commonMark
 	 *
-	 * @return string
+	 * @return \GrahamCampbell\Markdown\Responses\MarkdownString
```
Comment (GrahamCampbell) [inline src/Converters/Converter.php]: Unfortunately, we can't just change this type, because it now violates the interface.

Comment (GrahamCampbell) [reply src/Converters/Converter.php]: This is really a limitation of PHP. :/

Comment (GrahamCampbell) [reply src/Converters/Converter.php]: What we really want here is an instance of `Stringish`, like in hacklang.

**Comment by GrahamCampbell** [comment]
With the interface thing, I didn't mean just change it I;m afraid. I was highlighting that we actually have to return a string here, if we want to implement that interface.

**Comment by megaxlr** [comment]
Take a look at the constructor for MarkdownString and take a look at e() in
helpers.php in Laravel. It runs toHtml() if someone uses {{}} instead of
{!! !!}. That's the only thing in that interface.

On 8 Dec 2017 23:13, "Graham Campbell" <notifications@github.com> wrote:

With the interface thing, I didn't mean just change it I;m afraid. I was
highlighting that we actually have to return a string here, if we want to
implement that interface.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/GrahamCampbell/Laravel-Markdown/pull/96#issuecomment-350385406>,
or mute the thread

**Comment by GrahamCampbell** [comment]
The `League\CommonMark\Converter` interface says that method must return a `string`.

**Comment by megaxlr** [comment]
How about a [Class decorator](https://en.wikipedia.org/wiki/Decorator_pattern)? It wouldnt violate the interface. and an IDE would be happy to use it. 

In any case, It's less hacky than the current solution.

File: src/Converters/Converter.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+use GrahamCampbell\Markdown\Responses\MarkdownString;
+
+class Converter extends \League\CommonMark\Converter {
+	
+	/**
+	 * Converts CommonMark to HTML.
+	 *
+	 * @param string $commonMark
+	 *
+	 * @return \GrahamCampbell\Markdown\Responses\MarkdownString
```
Comment (GrahamCampbell) [inline src/Converters/Converter.php]: This still violates the interface it extends.

**Comment by GrahamCampbell** [comment]
Thanks, but I don't think I want to change anything here.

**Pull request closed without merging**